src/mlpack/methods/ann/rbm.hpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file rbm.hpp
+ * @author Kris Singh
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_RBM_HPP
+#define MLPACK_METHODS_ANN_RBM_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+#include <mlpack/methods/ann/activation_functions/softplus_function.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial neural networks.  */ {
+
+template<typename InitializationRuleType, typename RBMPolicy>
+class RBM
+{
+ public:
+  using NetworkType = RBM<InitializationRuleType, RBMPolicy>;
+  typedef typename RBMPolicy::ElemType eT;
+
+  /* 
+   * Intalise all the parameters of the network
+   * using the intialise rule. 
+   *
+   * @tparam IntialiserType rule to intialise the parameters of the network
+   * @param predictors training data
+   * @param numSteps Number of gibbs steps sampling
+   * @param useMonitoringCost evaluation function to use
+   * @param persistence indicates to use persistent CD
+   */
+  RBM(arma::Mat<eT> predictors, InitializationRuleType initializeRule,
+      RBMPolicy rbmPolicy,
+      const size_t numSteps = 1,
+      const size_t mSteps = 1,
+      const bool useMonitoringCost = true,
+      const bool persistence = false);
+
+  // Reset the network
+  void Reset();
+
+  /* 
+   * Train the network using the Opitimzer with given set of args.
+   * the optimiser sets the parameters of the network for providing
+   * most likely parameters given the inputs
+   * @param: predictors data points
```

Comment:
Also can you use  proper grammar and punctuation, for more information take a look at: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#comments in most cases this means adding ``.`` at the end of the line and to start with an upper-case letter.